### PR TITLE
update HALO-0126 to new definitions

### PR DESCRIPTION
**Pull request by Geet-George**

### CODE REVIEW

**Comment by heikekonow** [comment]
Hi Geet! I think you removed the file flight_phase_files/HALO_RF04_20200126_info.yaml in your last commit. Could you add it again?

**Comment by Geet-George** [comment]
Sorry.. That was stupid. I wanted to remove the notebooks for RF03 and RF04, but I removed the YAML instead. The mistakes have been corrected.

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 1]
```diff
@@ -21,71 +26,366 @@ remarks:
 - Four coordinated straight legs over Meteor.
 - Cirrus clouds in 14-17 km present for almost full flight.
 segments:
-- kind: straight_leg
+- kinds:
+  - straight_leg
   name: super_curtain
+  irregularities: []
+  segment_id: HALO-0126sc1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: Segment ID should combine flight id with underscore: HALO-0126sc1 -> HALO-0126_sc1

Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_sc1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 2]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 12:17:10
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-1
+  irregularities: []
+  segment_id: HALO-0126c1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 3]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s10
+    - HALO-0126_s11
+    - HALO-0126_s12
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-1 and circle-2
+  irregularities: []
+  segment_id: HALO-0126cb1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_cb1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 4]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 13:34:10
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-2
+  irregularities: []
+  segment_id: HALO-0126c2
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c2
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 5]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s24
+    BAD:
+    - HALO-0126_s19
+    UGLY:
+    - HALO-0126_s13
+- kinds:
+  - circle_break
+  name: Circle break between circle-2 and circle-3
+  irregularities: []
+  segment_id: HALO-0126cb2
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_cb2
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 6]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 14:42:24
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-3
+  irregularities: []
+  segment_id: HALO-0126c3
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c3
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 7]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s34
+    - HALO-0126_s36
+    BAD: []
+    UGLY:
+    - HALO-0126_s35
+- kinds:
+  - circling
+  name: Circling phase 1 - Circle 1 to 3
+  irregularities: []
+  segment_id: HALO-0126o1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_o1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 8]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD: []
+    BAD: []
+    UGLY:
+    - HALO-0126_s37
+- kinds:
+  - straight_leg
+  name: Leg-2 over Meteor
+  irregularities:
+  - slightly high roll at start of leg
+  segment_id: HALO-0126sl2
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_sl2
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 9]
```diff
@@ -21,71 +26,366 @@ remarks:
...
-  name: leg3_over_Meteor
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - straight_leg
+  name: Leg-3 over Meteor
+  irregularities: []
+  segment_id: HALO-0126sl3
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_sl3
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 10]
```diff
@@ -21,71 +26,366 @@ remarks:
...
-  name: leg4_over_Meteor
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - straight_leg
+  name: Leg-4 over Meteor
+  irregularities: []
+  segment_id: HALO-0126sl4
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_sl4
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 11]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD:
+    - HALO-0126_s38
+    - HALO-0126_s39
+    BAD: []
+    UGLY: []
+- kinds:
+  - radar_calibration_wiggle
+  name: Radar Calibration Wiggle
+  irregularities: []
+  segment_id: HALO-0126rcw1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_rcw1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 12]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 17:16:15
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-4
+  irregularities: []
+  segment_id: HALO-0126c4
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c4
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 13]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s49
+    - HALO-0126_s50
+    - HALO-0126_s51
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-4 and circle-5
+  irregularities: []
+  segment_id: HALO-0126cb3
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_cb3
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 14]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 18:34:59
+  dropsondes:
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-5
+  irregularities: []
+  segment_id: HALO-0126c5
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c5
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 15]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s63
+    BAD:
+    - HALO-0126_s55
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-5 and circle-6
+  irregularities:
+  - flight deviated from circle path due to traffic reasons
+  segment_id: HALO-0126cb4
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_cb4
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 16]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-6
+  irregularities:
+  - flight returned towards the island before completing a full circle
+  - circle end time denoted by change in altitude after last sonde
+  segment_id: HALO-0126c6
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_c6
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 17]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s74
+    - HALO-0126_s75
+    BAD: []
+    UGLY: []
+- kinds:
+  - circling
+  name: Circling phase 2 - Circle 4 to 6
+  irregularities:
+  - between circle 5 and 6, the flight deviated from the circle due to a traffic issue
+  segment_id: HALO-0126o2
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_o2
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 18]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s75
+    BAD:
+    - HALO-0126_s55
+    UGLY: []
+- kinds:
+  - lidar_leg
+  - straight_leg
+  name: Low-level lidar leg
+  irregularities: []
+  segment_id: HALO-0126ll1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_ll1
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 19]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s19
+    UGLY:
+    - HALO-0126_s13
+    - HALO-0126_s35
+- kinds:
+  - straight_leg
+  name: Leg-1 over Meteor
+  irregularities:
+  - slightly high roll towards end of the leg
+  segment_id: HALO-0126sl1
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  segment_id: HALO-0126_sl1
```

**Review by heikekonow — CHANGES_REQUESTED** [review]

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 20]
```diff
@@ -21,71 +26,366 @@ remarks:
 - Four coordinated straight legs over Meteor.
 - Cirrus clouds in 14-17 km present for almost full flight.
 segments:
-- kind: straight_leg
+- kinds:
+  - straight_leg
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:32]: This segment should also be kind "super_curtain"
```suggestion
  - straight_leg
  - super_curtain
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 21]
```diff
@@ -5,12 +5,17 @@ flight_id: HALO-0126
 contacts:
 - name: Marc Prange
   email: marc.prange@uni-hamburg.de
-  tags:
-  - dataset creator
+  tags:
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: There were some characters behind "tags:" that caused the reports scrip to provide an error message
```suggestion
  tags:
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 22]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-1
+  irregularities: []
+  segment_id: HALO-0126c1
+  start: 2020-01-26 12:26:08
+  end: 2020-01-26 13:22:28
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: Start and end overlap a bit; end circle two seconds earlier
```suggestion
  end: 2020-01-26 13:22:26
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 23]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s11
+    - HALO-0126_s12
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-1 and circle-2
+  irregularities: []
+  segment_id: HALO-0126cb1
+  start: 2020-01-26 13:22:29
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: Change start according to end of c1
```suggestion
  start: 2020-01-26 13:22:26
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 24]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-3
+  irregularities: []
+  segment_id: HALO-0126c3
+  start: 2020-01-26 14:42:25
+  end: 2020-01-26 15:39:13
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: This circle can be closed if it ends three seconds later
```suggestion
  end: 2020-01-26 15:39:16
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 25]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    BAD: []
+    UGLY:
+    - HALO-0126_s35
+- kinds:
+  - circling
+  name: Circling phase 1 - Circle 1 to 3
+  irregularities: []
+  segment_id: HALO-0126o1
+  start: 2020-01-26 12:26:08
+  end: 2020-01-26 15:39:13
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: This circling period can be extended a bit in beginning and end for as long as HALO is on the circle pattern.
```suggestion
  start: 2020-01-26 12:24:06
  end: 2020-01-26 15:46:12
```

**Comment by Geet-George** [comment]
For circle endings, are you looking at the zoomed-in plot? I went for the timestamp closest to a 360 degree completion, using the "Difference in heading" from the HTML that Tobi's report.py provided....

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 26]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    GOOD: []
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle
+  name: Circle-4
+  irregularities: []
+  segment_id: HALO-0126c4
+  start: 2020-01-26 17:19:54
+  end: 2020-01-26 18:18:34
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:244]: This circle overlaps. The overlap can be reduced by ending the circle earlier.
```suggestion
  end: 2020-01-26 18:17:54
```

Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: This circle overlaps. The overlap can be reduced by ending the circle earlier.
```suggestion
  end: 2020-01-26 18:17:54
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 27]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s50
+    - HALO-0126_s51
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-4 and circle-5
+  irregularities: []
+  segment_id: HALO-0126cb3
+  start: 2020-01-26 18:18:35
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: If the end of c4 is changed, then the start of cb3 should be changed accordingly. 
```suggestion
  start: 2020-01-26 18:17:54
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 28]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s12
+    BAD: []
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-1 and circle-2
+  irregularities: []
+  segment_id: HALO-0126cb1
+  start: 2020-01-26 13:22:29
+  end: 2020-01-26 13:34:10
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]: End time of the circle break can be the same as the start time of the following circle, as we decided that the end is not part of the second (and vice versa).
```suggestion
  end: 2020-01-26 13:34:11
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 29]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    BAD:
+    - HALO-0126_s19
+    UGLY:
+    - HALO-0126_s13
+- kinds:
+  - circle_break
+  name: Circle break between circle-2 and circle-3
+  irregularities: []
+  segment_id: HALO-0126cb2
+  start: 2020-01-26 14:30:32
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:105]:
```suggestion
  start: 2020-01-26 14:30:31
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 30]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  end: 2020-01-26 14:42:24
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  end: 2020-01-26 14:42:25
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 31]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s60
+    - HALO-0126_s61
+    - HALO-0126_s62
+    - HALO-0126_s63
+    BAD:
+    - HALO-0126_s55
+    UGLY: []
+- kinds:
+  - circle_break
+  name: Circle break between circle-5 and circle-6
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:300]: End circle break with deviation due to traffic.
```suggestion
  name: Circle break between circle-5 and deviation due to traffic
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 32]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  irregularities:
+  - flight deviated from circle path due to traffic reasons
+  segment_id: HALO-0126cb4
+  start: 2020-01-26 19:33:17
+  end: 2020-01-26 19:50:30
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  start: 2020-01-26 19:33:16
  end: 2020-01-26 19:40:48
```
Change in start time is set to end of previous circle (I missed this one in the other comments)

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 33]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s71
+    - HALO-0126_s72
+    - HALO-0126_s73
+    - HALO-0126_s74
+    - HALO-0126_s75
+    BAD: []
+    UGLY: []
+- kinds:
+  - circling
+  name: Circling phase 2 - Circle 4 to 6
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  name: Circling phase 2 - Circle 4 to deviation due to traffic
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 34]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+  irregularities:
+  - between circle 5 and 6, the flight deviated from the circle due to a traffic issue
+  segment_id: HALO-0126o2
+  start: 2020-01-26 17:19:54
+  end: 2020-01-26 20:47:00
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml]:
```suggestion
  end: 2020-01-26 19:40:48
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 35]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    - HALO-0126_s66
+    - HALO-0126_s67
+    - HALO-0126_s68
+    - HALO-0126_s69
+    - HALO-0126_s70
+    - HALO-0126_s71
+    - HALO-0126_s72
+    - HALO-0126_s73
+    - HALO-0126_s74
+    - HALO-0126_s75
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:392]: These will be moved to new segment o3
```suggestion
```

File: flight_phase_files/HALO_RF04_20200126_info.yaml [hunk 36]
```diff
@@ -21,71 +26,366 @@ remarks:
...
+    BAD:
+    - HALO-0126_s55
+    UGLY: []
+- kinds:
```
Comment (heikekonow) [inline flight_phase_files/HALO_RF04_20200126_info.yaml:395]:
```suggestion
- kinds:
  - circling
  name: Circling phase 3 - Deviation due to traffic to circle 6
  irregularities:
  - between circle 5 and 6, the flight deviated from the circle due to a traffic issue
  segment_id: HALO-0126_o3
  start: 2020-01-26 19:51:19
  end: 2020-01-26 20:47:00
  dropsondes:
    GOOD:
    - HALO-0126_s64
    - HALO-0126_s65
    - HALO-0126_s66
    - HALO-0126_s67
    - HALO-0126_s68
    - HALO-0126_s69
    - HALO-0126_s70
    - HALO-0126_s71
    - HALO-0126_s72
    - HALO-0126_s73
    - HALO-0126_s74
    - HALO-0126_s75
    BAD: []
    UGLY: []
- kinds:
```

**Review by heikekonow — COMMENTED** [review]
Tried to update o2 and create o3 to split the circling elements around the deviation due to traffic

**Comment by heikekonow** [comment]
Ok, I hope I filled in all my notes. 
Overall, this looks great. I have only a couple of minor comments and the deviation due to traffic as we discussed.
Thanks for the segmentation of this flight. That was a tricky one :-)

**Comment by heikekonow** [comment]
> For circle endings, are you looking at the zoomed-in plot? I went for the timestamp closest to a 360 degree completion, using the "Difference in heading" from the HTML that Tobi's report.py provided....

Yes, I used the zoomed in plots. Going for difference in heading will probably also work. I'll leave the decision up to you.

**Comment by d70-t** [comment]
> > For circle endings, are you looking at the zoomed-in plot? I went for the timestamp closest to a 360 degree completion, using the "Difference in heading" from the HTML that Tobi's report.py provided....
> 
> Yes, I used the zoomed in plots. Going for difference in heading will probably also work. I'll leave the decision up to you.

Note that heading will also change due to changes in wind speed and direction. So completing a circle does not necessarily mean returning to the same heading.

**Comment by Geet-George** [comment]
But sticking to the rules (definition), shouldn't we go by least difference in heading? That was my thought process... I did see in the zoomed-in plots that doing so showed more or less overlap in the circle than we would like...

**Comment by d70-t** [comment]
> But sticking to the rules (definition), shouldn't we go by least difference in heading? That was my thought process... I did see in the zoomed-in plots that doing so showed more or less overlap in the circle than we would like...

Hmm, I think I begin to see your point. In my head the idea of a 360° circle never has been 360° of heading change but 360° of rotation around the center of the circle and the heading being only a rough estimate for that number. Maybe our definition has been a bit unclear on that an we should figure that out?

**Comment by Geet-George** [comment]
It was also rotation of 360° in my head, but I thought we use heading as the measurement to estimate that, and that's why I stuck to the definition, partly also because I thought marking circle completion at the return point closest to the start, will not be easy to define.

**Comment by Geet-George** [comment]
Okay, since this idea of least heading difference is the odd one out, for convenience sake, I'll just change the times considering overlap of the circle. We can change the final definition accordingly later.

**Comment by Geet-George** [comment]
Timestamps updated to minimise circle overlap, without considering heading differences. 

RF04 is now ready to merge.

**Pull request merged**